plotter/step.go

```diff
@@ -0,0 +1,168 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image/color"
+	"math"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+)
+
+// StepKind specifies a form of a connection of two consecutive points.
+type StepKind int
+
+const (
+	// PreStep means that two consecutive points are connected by two lines in the following order: vertical, horizontal.
```

Comment:
Done. Before approval... Do you think it is worth to merge `Step` with `Line`? I wrote about this in the top comment. Basically we can just add `NoStep` default to `Line`. Seems logical to me. Line stays line but we can control its shape in this case.

Reply:
I'll leave that up to @sbinet.

Reply:
I think it would be interesting to see what it would look like.

Reply:
maybe @ctessum has something to say about this as well?

Reply:
@sbinet I can add another commit to this pull request to show what it would look like. If we decide that separate `Step` is better then I'll just revert it. I'm busy right now, so I'll do it in 1 or 2 days.

Reply:
SGTM. Thanks